docs/Serverless.md

```diff
@@ -0,0 +1,82 @@
+<h1 align="center">Serverless</h1>
+
+Run serverless applications and REST APIs using your existing fastify application.
+
+
+## AWS Lambda
+
+On top of AWS Lambda and Amazon API Gateway. The sample provided allows you to easily build serverless web applications/services and RESTful APIs using fastify.
+
+
+Having a normal fastify application like this:
+
+```js
+const fastify = require('fastify');
+
+const app = fastify({ serverFactory });
+
+app.get('/', (request, reply) => reply.send({ hello: 'world' }));
+
+app.listen(3000, () => console.log('server listening on 3000'));
+```
+
+*Note: This is just one possible way.*
+
+You can simply wrap your initialization code by offering to inject an optional [serverFactory](https://www.fastify.io/docs/latest/Server/#serverfactory).
+
+When executed in your lambda function we don't need to listen to a specific port, so we just export the wrapper function `init` in this case. The [`lambda.js`](https://www.fastify.io/docs/latest/Server/#lambda.js) file will use this export.
+
+When you execute your fastify application like always, i.e. `node app.js` *(the detection for this could be `require.main === module`)*, you can normally listen to your port, so you can still run your fastify function locally.
+
```

Comment:
I really don't understand any of the preceding text. It doesn't seem to have any relation to the code in this subsection of the document.

Reply:
moved the text below the code section